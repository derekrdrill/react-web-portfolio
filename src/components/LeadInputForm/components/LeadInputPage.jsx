import React, { useContext } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { AppBar, Toolbar, Button, Tooltip, Typography } from '@mui/material';
import { LeadInputContext } from '../context/LeadInputContext';
import { LeadInputForm } from './LeadInputForm';
import { LeadInputDataTable } from './LeadInputDataTable';

export const LeadInputPage = () => {
  const { leadInputDispatch, page, tooltipOpen } = useContext(LeadInputContext);

  const handlePageButtonClick = () => {
    const newPage = page === 'form' ? 'table' : 'form';
    leadInputDispatch({ type: 'SWITCH_PAGES', payload: { page: newPage, tooltipOpen: false } });
  };

  const handleTooltipToggle = () => leadInputDispatch({ type: 'TOGGLE_TOOLTIP', payload: !tooltipOpen });

  return (
    <>
      <PageBodyStyle />
      <LeadInputPageHeader>
        <Toolbar>
          <Container fluid>
            <Row>
              <ButtonContainer xs={12}>
                <Tooltip
                  open={tooltipOpen}
                  onClose={handleTooltipToggle}
                  onOpen={page === 'form' && handleTooltipToggle}
                  title={
                    <Typography variant='subtitle2' component='p'>
                      This would likely be an admin button in a real-life scenario
                    </Typography>
                  }
                >
                  <Button onClick={handlePageButtonClick}>
                    {page === 'form' ? 'See all records' : 'Go back to input form'}
                  </Button>
                </Tooltip>
              </ButtonContainer>
            </Row>
          </Container>
        </Toolbar>
      </LeadInputPageHeader>
      <LeadInputPageContent>
        {page === 'form' && <LeadInputForm />}
        {page === 'table' && <LeadInputDataTable />}
      </LeadInputPageContent>
    </>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: 'tan',
  },
});

const LeadInputPageHeader = styled(AppBar)({
  top: 80,
  boxShadow: 'none',
  backgroundColor: 'inherit',
});

const LeadInputPageContent = styled.div({
  marginTop: '10%',
});

const ButtonContainer = styled(Col)({
  textAlign: 'right',
});
