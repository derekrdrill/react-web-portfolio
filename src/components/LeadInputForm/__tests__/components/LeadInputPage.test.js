import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import LeadInputPage from '../../components/LeadInputPage';

const renderer = new ShallowRenderer();

describe('Lead Input Page tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({
      darkMode: true,
      leadInputDispatch: jest.fn(),
      page: 1,
      tooltipOpen: false,
    });

    renderer.render(<LeadInputPage />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
