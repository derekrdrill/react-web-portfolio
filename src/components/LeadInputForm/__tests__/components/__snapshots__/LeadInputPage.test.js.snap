// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lead Input Page tests renders correctly 1`] = `
<React.Fragment>
  <Memo(l)
    darkMode={true}
  />
  <ForwardRef(Grid)
    container={true}
  >
    <Styled(Component)
      item={true}
      xs={12}
    >
      <ForwardRef(Tooltip)
        onClose={[Function]}
        onOpen={false}
        open={false}
        title={
          <ForwardRef(Typography)
            component="p"
            variant="subtitle2"
          >
            This would likely be an admin button in a real-life scenario
          </ForwardRef(Typography)>
        }
      >
        <ForwardRef(Button)
          onClick={[Function]}
        >
          Go back to input form
        </ForwardRef(Button)>
      </ForwardRef(Tooltip)>
    </Styled(Component)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    container={true}
    style={
      Object {
        "paddingTop": 75,
      }
    }
  />
</React.Fragment>
`;
